deepparse/comparer/address_comparer.py

```diff
@@ -0,0 +1,170 @@
+from deepparse.parser import AddressParser
+from typing import List, Union
+
+
+class AdressComparer:
+    """
+        Compares addresses with each others and retrieve the the differences between them.
+    """
+    
+    def __init__(self, parser: AddressParser) -> None:
+        """
+        Address parser used to parse the addresses
+        """
+
+        self.parser = parser
+
+    def __str__(self) -> str:
+        return f"Compare addresses with {self.parser.model_type.capitalize()}AddressParser"
+
+    __repr__ = __str__  # to call __str__ when list of address
+    
+    def compare_with_deepparse(self, addresses_to_compare: Union[List[List[tuple]], List[tuple]]) -> dict:
+        """
+        Compare a list of already parsed addresses and then compare with the result 
+        it would be if it was parsed using deepparse 
+
+        Args:
+            addresses_to_compare (Union[List[List[tuple]], List[tuple]]): List of tuple where
+            the first element in the tuple is the value of the address component, and the 
+            second element is the name of the address component
+
+        Return:
+            Dictionnary that contains dictionnaries that contains all addresses components that differ from the original
+            parsing and the deepparsed components
+        """
+        if isinstance(addresses_to_compare[0], tuple):  # when tag is also the tag and the probability of the tag
+            addresses_to_compare = [addresses_to_compare]
```

Comment:
Not sure to understand. What if a list of lists?

Reply:
if it is a list of list, then you can already iterate of the addresses, otherwise the list of tuple is appended to a list so the rest of the code works the same